x/poolmanager/protorev.go

```diff
@@ -1,91 +1,126 @@
 package poolmanager
 
 import (
+	"fmt"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	gogotypes "github.com/cosmos/gogoproto/types"
 
+	"github.com/osmosis-labs/osmosis/osmomath"
 	"github.com/osmosis-labs/osmosis/osmoutils"
 	"github.com/osmosis-labs/osmosis/v21/x/poolmanager/types"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
 )
 
-// IncreaseTakerFeeTrackerForStakers gets the current value of the taker fee tracker for stakers, adds the given amount to it, and sets the new value.
-func (k Keeper) IncreaseTakerFeeTrackerForStakers(ctx sdk.Context, takerFeeForStakers sdk.Coin) {
-	currentTakerFeeForStakers := k.GetTakerFeeTrackerForStakers(ctx)
-	if !takerFeeForStakers.IsZero() {
-		newTakerFeeForStakersCoins := currentTakerFeeForStakers.Add(takerFeeForStakers)
-		newTakerFeeForStakers := types.TrackedVolume{
-			Amount: newTakerFeeForStakersCoins,
+// GetTakerFeeTrackerForStakers returns the taker fee for stakers tracker for all denoms that has been
+// collected since the accounting height.
+func (k Keeper) GetTakerFeeTrackerForStakers(ctx sdk.Context) []sdk.Coin {
```

Comment:
The Getters, Getters by Denom, and Update logic follows the same pattern that protorev uses for tracking profits by denom.